src/argparse.lua

```diff
@@ -1087,6 +1087,230 @@ function Parser:get_help()
+         print(self["get_" .. shell .. "_complete"](self))
+         os.exit(0)
+      end)
+
+   if value then
+      complete = complete(value)
+   end
+
+   if not complete._name then
+      complete "--completion"
+   end
+
+   return self
+end
+
+local function get_short_description(element)
+   local short = element._description:match("^(.-)%.%s")
+   return short or element._description:match("^(.-)%.?$")
+end
+
+local function get_options(parser)
+   local options = {}
+   for _, option in ipairs(parser._options) do
+      for _, alias in ipairs(option._aliases) do
+         table.insert(options, alias)
+      end
+   end
+   return table.concat(options, " ")
+end
+
+local function get_commands(parser)
+   local commands = {}
+   for _, command in ipairs(parser._commands) do
+      for _, alias in ipairs(command._aliases) do
+         table.insert(commands, alias)
+      end
+   end
+   return table.concat(commands, " ")
+end
+
+function Parser:_bash_option_args(buf, indent)
+   local opts = {}
+   for _, option in ipairs(self._options) do
+      if option._choices or option._minargs > 0 then
+         local pattern = ("%s)"):format(table.concat(option._aliases, "|"))
+         local compreply
+         if option._choices then
+            compreply = ('COMPREPLY=($(compgen -W "%s" -- "$cur"))')
+               :format(table.concat(option._choices, " "))
+         else
+            compreply = 'COMPREPLY=($(compgen -f "$cur"))'
+         end
+         table.insert(opts, (" "):rep(indent + 4) .. pattern)
+         table.insert(opts, (" "):rep(indent + 8) .. compreply)
+         table.insert(opts, (" "):rep(indent + 8) .. "return 0")
+         table.insert(opts, (" "):rep(indent + 8) .. ";;")
+      end
+   end
+
+   if #opts > 0 then
+      table.insert(buf, (" "):rep(indent) .. 'case "$prev" in')
+      table.insert(buf, table.concat(opts, "\n"))
+      table.insert(buf, (" "):rep(indent) .. "esac\n")
+   end
+end
+
+function Parser:_bash_get_cmd(buf)
+   local cmds = {}
+   for idx, command in ipairs(self._commands) do
+      if idx ~= self._help_command_idx then
+         table.insert(cmds, (" "):rep(12) .. ("%s)"):format(table.concat(command._aliases, "|")))
+         table.insert(cmds, (" "):rep(16) .. ('cmd="%s"'):format(command._aliases[1]))
+         table.insert(cmds, (" "):rep(16) .. "break")
+         table.insert(cmds, (" "):rep(16) .. ";;")
+      end
+   end
+
+   if #cmds > 0 then
+      table.insert(buf, (" "):rep(4) .. "for arg in ${COMP_WORDS[@]:1}")
+      table.insert(buf, (" "):rep(4) .. "do")
+      table.insert(buf, (" "):rep(8) .. 'case "$arg" in')
+      table.insert(buf, table.concat(cmds, "\n"))
+      table.insert(buf, (" "):rep(8) .. "esac")
+      table.insert(buf, (" "):rep(4) .. "done\n")
+   end
+end
+
+function Parser:_bash_cmd_completions(buf)
+   local subcmds = {}
+   for idx, command in ipairs(self._commands) do
+      if #command._options > 0 and idx ~= self._help_command_idx then
```

Comment:
Why exclude the help command here?

Reply:
The help command is excluded in the for loop built by `_bash_get_cmd` so that it uses the completions for the main command (so that it completes subcommands).